internal/api/v1/handlers/instance.go

```diff
@@ -156,3 +156,127 @@ func (h *InstanceHandler) GetInstance(c *fiber.Ctx) error {
 
 	return c.JSON(instance)
 }
+
+// GetPublicIPs returns a list of public IPs and their associated job IDs
+func (h *InstanceHandler) GetPublicIPs(c *fiber.Ctx) error {
```

Comment:
This would need to have pagination as the instances can go high.
Same for a number of other endpoints